spec/cspec/c/kernel.mk

```diff
@@ -115,3 +115,16 @@ ${KERNEL_BUILD_ROOT}/.cmake_done: ${KERNEL_DEPS} ${CONFIG_DOMAIN_SCHEDULE}
 
 ${UMM_TYPES}: ${KERNEL_BUILD_ROOT}/kernel_all.c_pp
 	${CSPEC_DIR}/mk_umm_types.py --root $(L4V_REPO_PATH) ${KERNEL_BUILD_ROOT}/kernel_all.c_pp $@
+
+# This target generates config files and headers only. It does not invoke
+# the C tool chain or preprocessor. We force CMake to skip tests for these,
+# so that ASpec and ExecSpec can be built with fewer dependencies.
```

Comment:
out of curiosity: how did you manage this, via no-compiler.cmake? how do you stop it trying to build sources?

Reply:
`cmake` itself doesn't build, it's more like `autoconf` and friends, it just generates info for another build system. That other build system could be `make` for instance, but for seL4 it's `ninja`. For this one we just never call `ninja`.

The one remaining bit is that `cmake` tests for compiler availability when it does all it's build system generation stuff, and the `no-compiler.cmake` settings force it to skip most of that. It will still check that *some* compiler is installed, but it will not demand any more that it works or that it works for the selected platform.

Reply:
Great. The part I missed is that it's cmake that generates the config, and not a part of a target in a Makefile-analogue (in this case ninja). Now I got it.